### PR TITLE
cwr : add SWR,OWR records and additional SPU fields.

### PR DESCRIPTION
**Pull request by orenyodfat**
- Add SWR and OWR tx records.
- Additional  fields to SPU record.

- Add contributors field to graphql query so a graphql query for a registered_work with iscw 'X' will look like : 
```
{
      registered_work(iswc: "X") {
        iswc 
        title
        contributors{
          writer_first_name
          writer_last_name
          writer_ipi_name
          writer_ipi_base_number 
        }
      }
    }
```

### CODE REVIEW

File: cwr/schema.go [hunk 1]
```diff
@@ -57,21 +57,67 @@ CREATE INDEX registered_work_cwr_id_idx         ON registered_work (cwr_id);
...
+	mr_ownership_share                text NOT NULL,
+	sr_society                        text NOT NULL,
+	sr_ownership_share                text NOT NULL,
+	special_agreements_indicator      text NOT NULL,
+	first_recording_refusal_ind       text NOT NULL,
+	publisher_ipi_base_number         text NOT NULL,
+	inter_standard_agreement_code     text NOT NULL,
+	society_assigned_agreement_number text NOT NULL,
+	agreement_type                    text NOT NULL,
+	usa_license_ind                   text NOT NULL,
```
Comment (lmars) [inline cwr/schema.go]: Do we really need to index on all of these fields?

File: cwr/api_test.go [hunk 2]
```diff
@@ -217,6 +261,22 @@ func testTxRecords(x *testIndex,
...
+				obj, err := x.store.Get(swrCid.(*cid.Cid))
+				if err != nil {
+					return err
+				}
+				if err := obj.Decode(record); err != nil {
+					return err
+				}
+				if record.RecordType == "SWR" {
+					if record.WriterFirstName != "" {
+						if err := assertQuerySWR(record, `{ writer_control(writer_first_name:%q) { writer_first_name } }`, record.WriterFirstName); err != nil {
```
Comment (lmars) [inline cwr/api_test.go]: We should test a top level query like getting a works and all its contributors in one go.

File: cwr/convert.go [hunk 3]
```diff
@@ -228,6 +234,39 @@ func newRecord(line string) (*Record, error) {
...
+		record.SocietyAssignedAgreementNumber = substring(line, 166, 180)
+		record.AgreementType = substring(line, 180, 182)
+		record.USALicenseInd = substring(line, 182, 183)
+	case "SWR", "OWR":
+		record.RecordType = substring(line, 0, 3)
+		record.TransactionSequenceN = substring(line, 3, 12)
+		record.RecordSequenceN = substring(line, 12, 19)
+		record.InterestedPartyNumber = substring(line, 19, 28)
+		record.WriterLastName = substring(line, 28, 73)
+		record.WriterFirstName = substring(line, 73, 103)
```
Comment (lmars) [inline cwr/convert.go]: Should these strings be trimmed of trailing whitespace?

File: cwr/schema.go (hunk 1 above)
Comment (orenyodfat) [reply cwr/schema.go]: why not ?

File: cwr/convert.go (hunk 3 above)
Comment (orenyodfat) [reply cwr/convert.go]: I guess it should be trimmed like all others fields

File: cwr/api_test.go (hunk 2 above)
Comment (orenyodfat) [reply cwr/api_test.go]: we have it on line 244 though might need to add all fields

File: cwr/schema.go (hunk 1 above)
Comment (lmars) [reply cwr/schema.go]: The more things we index, the longer it takes and the bigger the resulting index is.

For example, we should only index on `usa_license_ind` if we know we want to query on that field, but I don't see a corresponding query so why index it at all?

File: cwr/api_test.go (hunk 2 above)
Comment (orenyodfat) [reply cwr/api_test.go]: update at 7d2774d

File: cwr/convert.go (hunk 3 above)
Comment (orenyodfat) [reply cwr/convert.go]: update at 7d2774d

**Comment by orenyodfat** [comment]
@lmars thanks for your comments

**Pull request merged**